src/tests/super-admins.test.js

```diff
@@ -1 +1,150 @@
-test.skip('skip', () => {});
+/* eslint-disable no-underscore-dangle */
+import request from 'supertest';
+import app from '../app';
+import superAdmins from '../models/super-admins';
+import superAdminSeed from '../seeds/super-admins';
+
+beforeEach(async () => {
+  await superAdmins.collection.insertMany(superAdminSeed);
+});
+
+const mockSuperAdmin = {
+  firstName: 'Micaela',
+  email: 'micaela@prueba.com',
+  password: 'Pr0band*',
+};
+
+const mockSuperAdminLessName = {
+  firstName: '',
+  email: 'test@test.com',
+  password: 'Pr0band&',
+};
+
+const mockSuperAdminLessEmail = {
+  firstName: 'James',
+  email: '',
+  password: 'Pr0band&',
+};
+
+const mockSuperAdminLessPassword = {
+  firstName: 'James',
+  email: 'some@email.com',
+  password: '',
+};
+
+const mockSuperAdminPasswordError = {
+  firstName: 'Zoe',
+  email: 'zoeee@email.com',
+  password: 'password',
+};
+
+afterEach(async () => {
+  await superAdmins.collection.deleteMany();
+});
+
+describe('GET /api/super-admins', () => {
+  test('If are super admins in data base status must be 200', async () => {
+    const response = await request(app).get('/api/super-admins').send();
+    expect(response.body.data.length).toBeGreaterThan(0);
+    expect(response.status).toBe(200);
+    expect(response.body.error).toBeFalsy();
+    expect(response.body).toBeDefined();
+    expect(response.body.message).toMatch('Super Admins list: ');
+  });
+  test('The status must be 404, if the url is wrong', async () => {
+    const response = await request(app).get('/api/superAdmins').send();
+    expect(response.status).toBe(404);
+    expect(response.body.error).toBeUndefined();
+  });
+  describe('GET /api/super-admins', () => {
+    test('Should return 404 if aren\'t super admins registered. And the message must be: Cannot find any Super Admin, please create one.', async () => {
+      await superAdmins.collection.deleteMany({});
+      const response = await request(app).get('/api/super-admins').send();
+      expect(response.body.data).toBe(undefined);
+      expect(response.status).toBe(404);
+      expect(response.body.message).toMatch('Cannot find any Super Admin, please create one');
+    });
+  });
+});
+
+describe('GET BY ID /api/super-admins/:id', () => {
+  test('Should return 200 if the id exist', async () => {
+    const response = await request(app).get(`/api/super-admins/${superAdminSeed[0]._id}`).send();
+    expect(response.status).toBe(200);
+    expect(response.body).toBeDefined();
+    expect(response.body.data).toHaveProperty('_id', 'firstName', 'email', 'password');
+    expect(response.body.message).toMatch(`Super Admin Found! ${superAdminSeed[0].firstName}`);
+    expect(response.body.error).toBeFalsy();
+  });
+  test('The status must be 404, if the url is wrong', async () => {
+    const response = await request(app).get('/api/superAdmins').send();
+    expect(response.status).toBe(404);
+    expect(response.body.error).toBeUndefined();
```

Comment:
be careful, in these cases the response doesn't have a ".body.error" so it doesn't make sense to test it for undefined

Reply:
Nai corrected these like that cause i told her the body error doesn't match to falsy or thruty. The test throws undefined